### ISSUE TITLE
Nav Unification: "domains" sub menu position not working

### ISSUE THREAD

**Issue by michaeldcain** [Automattic/wp-calypso#51150]
In the old Calypso navigation scheme, the Upgrades menu is:

- Upgrades
 - Plans
 - Domains
 - Purchases (where you manage both plan and domains purchases)

However, in the unified navigation it's:

- Upgrades
 - Plans
 - Purchases (where you manage both plan and domains purchases)
 - Domains

In `WPcom_Admin_Menu` the [position argument for the Domains sub menu](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/modules/masterbar/admin-menu/class-wpcom-admin-menu.php#L201) seems to be correct (10, where Plans is 5 and Purchases is 15), but isn't respected.

Long story short, it makes sense to have purchase management (Purchases) at the bottom of this sub-menu, especially as we add other items to the menu (i.e. email).

Reported in: p9Jlb4-2iV-p2

**Comment by obenland** [comment]
> In WPcom_Admin_Menu the position argument for the Domains sub menu seems to be correct (10, where Plans is 5 and Purchases is 15), but isn't respected.

Submenu items are kept in an unassociated array but [they get inserted at the specified key](https://core.trac.wordpress.org/browser/tags/5.7/src/wp-admin/includes/plugin.php#L1455). So switching the key to 2 or 3 or something should work to make it the third item in the list.

**Comment by tjcafferkey** [comment]
@hambai we are about to pick up this issue in our iteration and see you've opened a WIP PR to close this.

Is there any update around this?

**Comment by cpapazoglou** [comment]
@daledupreez mentioned that they will be picking this PR as part of their current iteration, since the email pages will probably be ready next week. convo: p1618862347352500-slack-CJS75TX3R

**Comment by daledupreez** [comment]
Note that as things stand, this change is currently rolled out to WPCOM Simple sites, and should make it across to WPCOM Atomic sites in Jetpack 9.7 early in May.

@Automattic/ajax, if you feel this is of sufficiently high priority to move across to WPCOM Atomic sites before the Jetpack update, please feel free to step in!

**Comment by cpapazoglou** [comment]
> Note that as things stand, this change is currently rolled out to WPCOM Simple sites, and should make it across to WPCOM Atomic sites in Jetpack 9.7 early in May.

That sounds enough! Thanks for working on this @daledupreez 👏
